File: AssetApplication.java
package com.example.demo;

import javafx.application.Application;
import javafx.event.EventHandler;
import javafx.fxml.FXMLLoader;
import javafx.scene.Parent;
import javafx.scene.Scene;
import javafx.scene.image.Image;
import javafx.stage.FileChooser;
import javafx.stage.Stage;
import javafx.stage.WindowEvent;
import java.io.IOException;
import java.io.PrintWriter;
import java.io.File;
import java.util.Scanner;

public class AssetApplication extends Application {
    @Override
    public void start(Stage stage) throws IOException {
        FileChooser fileChooser = new FileChooser();
        fileChooser.setTitle("Please choose Log File from folder");
        fileChooser.getExtensionFilters().add(new FileChooser.ExtensionFilter("CSV Files (*.csv)", "*.csv"));
        File file = fileChooser.showOpenDialog(stage);
        stage.getIcons().add(new Image(AssetApplication.class.getResourceAsStream("icon.png")));
        Parent root = FXMLLoader.load(AssetApplication.class.getResource("MainUI.fxml"));
        stage.setTitle("JMSDF ISR Wargame");
        stage.setScene(new Scene(root));
        stage.setResizable(true);
        readCSV(file);
        stage.setOnCloseRequest(new EventHandler<WindowEvent>() {
            public void handle(WindowEvent we) {
                writeCSV(file);
            }
        });

        stage.show();
    }

    public static void main(String[] args) {
        launch();
    }
    public void writeCSV(File file) {
        try {
            PrintWriter pw = new PrintWriter(file);
            StringBuilder lineBuilder = new StringBuilder();
            lineBuilder.append("Asset Name,");
            lineBuilder.append("Units Purchased,");
            lineBuilder.append("GCS/Launchers Purchased,");
            lineBuilder.append("Average Opposing Wind (kn),");
            lineBuilder.append("Cruise Altitude (ft),");
            lineBuilder.append("Availability Rate (*10),");
            lineBuilder.append("Range to Sweep Width Conversion (*10),");
            lineBuilder.append("Revisit Rate (hrs),");
            lineBuilder.append("Area of Regard (nm^2)");
            pw.write(lineBuilder + "\n");
            for(Asset asset : PurchasingBackend.cart.assetList) {
                lineBuilder = new StringBuilder();
                lineBuilder.append(asset.name + ",");
                lineBuilder.append(asset.UAVPurchased + ",");
                lineBuilder.append(asset.GCSPurchased + ",");
                lineBuilder.append(asset.averageOpposingWind + ",");
                lineBuilder.append(asset.cruiseAltitudeFt + ",");
                lineBuilder.append(asset.availabilityRate + ",");
                lineBuilder.append(asset.rangeToSweepWidthConversion + ",");
                lineBuilder.append(asset.revisitRate + ",");
                lineBuilder.append(asset.areaOfRegard + "");
                pw.write(lineBuilder + "\n");
            }
            pw.write("Initial Budget ($)," + PurchasingBackend.cart.initialBudget + "\n");
            pw.flush();
            pw.close();
        } catch (Exception e) {
            return;
        }
    }
    public void readCSV(File file) {
        try {
            Scanner csv = new Scanner(file);
            csv.nextLine();
            String[] line;
            for(Asset asset : PurchasingBackend.cart.assetList) {
                line = csv.nextLine().split(",");
                asset.UAVPurchased = Integer.parseInt(line[1]);
                asset.GCSPurchased = Integer.parseInt(line[2]);
                asset.averageOpposingWind = Integer.parseInt(line[3]);
                asset.cruiseAltitudeFt = Integer.parseInt(line[4]);
                asset.availabilityRate = Integer.parseInt(line[5]);
                asset.rangeToSweepWidthConversion = Integer.parseInt(line[6]);
                asset.revisitRate = Integer.parseInt(line[7]);
                asset.areaOfRegard = Integer.parseInt(line[8]);
            }
            line = csv.nextLine().split(",");
            PurchasingBackend.cart.initialBudget = Long.parseLong(line[1]);
        } catch (Exception e) {
            return;
        }
    }
}

File: python files/AssetApplication.py
import tkinter as tk
from tkinter import filedialog
import csv

class AssetApplication:
    def __init__(self):
        self.root = tk.Tk()
        self.root.title("JMSDF ISR Wargame")
        self.root.iconphoto(True, tk.PhotoImage(file="icon.png"))
        
        # Load UI from a separate file (assuming MainUI.py exists)
        from MainUI import create_ui
        create_ui(self.root)
        
        self.root.resizable(True, True)
        
        self.file = self.choose_file()
        if self.file:
            self.read_csv(self.file)
        
        self.root.protocol("WM_DELETE_WINDOW", self.on_closing)
        
    def choose_file(self):
        file = filedialog.askopenfilename(
            title="Please choose Log File from folder",
            filetypes=[("CSV Files", "*.csv")]
        )
        return file if file else None
    
    def on_closing(self):
        if self.file:
            self.write_csv(self.file)
        self.root.destroy()
    
    def write_csv(self, file):
        try:
            with open(file, 'w', newline='') as csvfile:
                writer = csv.writer(csvfile)
                writer.writerow([
                    "Asset Name", "Units Purchased", "GCS/Launchers Purchased",
                    "Average Opposing Wind (kn)", "Cruise Altitude (ft)",
                    "Availability Rate (*10)", "Range to Sweep Width Conversion (*10)",
                    "Revisit Rate (hrs)", "Area of Regard (nm^2)"
                ])
                for asset in PurchasingBackend.cart.assetList:
                    writer.writerow([
                        asset.name, asset.UAVPurchased, asset.GCSPurchased,
                        asset.averageOpposingWind, asset.cruiseAltitudeFt,
                        asset.availabilityRate, asset.rangeToSweepWidthConversion,
                        asset.revisitRate, asset.areaOfRegard
                    ])
                writer.writerow(["Initial Budget ($)", PurchasingBackend.cart.initialBudget])
        except Exception:
            pass
    
    def read_csv(self, file):
        try:
            with open(file, 'r') as csvfile:
                reader = csv.reader(csvfile)
                next(reader)  # Skip header
                for asset, row in zip(PurchasingBackend.cart.assetList, reader):
                    asset.UAVPurchased = int(row[1])
                    asset.GCSPurchased = int(row[2])
                    asset.averageOpposingWind = int(row[3])
                    asset.cruiseAltitudeFt = int(row[4])
                    asset.availabilityRate = int(row[5])
                    asset.rangeToSweepWidthConversion = int(row[6])
                    asset.revisitRate = int(row[7])
                    asset.areaOfRegard = int(row[8])
        except Exception:
            pass
    
    def run(self):
        self.root.mainloop()

if __name__ == "__main__":
    app = AssetApplication()
    app.run()

